src/snapshot_middleware/project.rs

```diff
@@ -23,7 +23,17 @@ pub fn snapshot_project(
 ) -> anyhow::Result<Option<InstanceSnapshot>> {
     let project = Project::load_from_slice(&vfs.read(path)?, path)
         .with_context(|| format!("File was not a valid Rojo project: {}", path.display()))?;
-    let project_name = project.name.as_deref().unwrap_or(name);
+    let project_name = match project.name.as_deref() {
+        Some(name) => name,
+        None => match path.file_name().and_then(OsStr::to_str) {
+            Some("default.project.json") => project
+                .folder_location()
+                .file_name()
+                .and_then(OsStr::to_str)
+                .unwrap_or(name),
+            _ => name,
+        },
+    };
```

Comment:
... and then, I replaced this with
```suggestion
    let project_name = match project.name.as_deref() {
        Some(name) => name,
        None => panic!("Project is missing a name"),
    };
```
Panicking is appropriate here, because if we ever load a project to snapshot and it's missing a name field, then it's a bug.

After these changes, optional name handling is entirely defined in the project module, and all tests pass. Are there any drawbacks you can see?